packages/terminology/src/seed/seedUmlsCrosswalk.ts

```diff
@@ -0,0 +1,352 @@
+  }
+}
+
+function createLookupParameters(system: string, code: string): Parameters {
+  return {
+    resourceType: "Parameters",
+    parameter: [
+      { name: "system", valueUri: system },
+      { name: "code", valueCode: code },
+    ],
+  };
+}
+
+async function createConceptMap({
+  concept,
+  config,
+  sourceDisplay,
+  targetDisplay,
+  targetCode,
+}: {
+  concept: UmlsConceptMap;
+  config: UmlsConceptMapConfig;
+  sourceDisplay: string | undefined;
+  targetDisplay: string | undefined;
+  targetCode?: string | undefined;
+}): Promise<ConceptMap> {
+  return {
+    resourceType: "ConceptMap",
+    status: "active",
+    group: [
+      {
+        source: config.system,
+        target: config.target,
+        element: [
+          {
+            code: concept.FROMID,
+            display: sourceDisplay,
+            target: [
+              {
+                code: targetCode ?? concept.TOID,
+                display: targetDisplay,
+                equivalence: "equivalent",
+              },
+            ],
+          },
+        ],
+      },
+    ],
+  };
+}
+
+function updateConceptMap({
+  umlsConcept,
+  conceptMap,
+  targetDisplay,
+  targetCode,
+}: {
+  umlsConcept: UmlsConceptMap;
+  conceptMap: ConceptMap;
+  targetDisplay: string | undefined;
+  targetCode?: string | undefined;
+}): ConceptMap {
+  return {
+    ...conceptMap,
+    group:
+      conceptMap.group?.map(group => ({
+        ...group,
+        element: group.element?.map(element =>
+          element.code === umlsConcept.FROMID
+            ? {
+                ...element,
+                target: [
+                  ...(element.target?.map(t => ({ ...t, equivalence: "narrower" as const })) ?? []),
+                  {
+                    code: targetCode ?? umlsConcept.TOID,
+                    display: targetDisplay,
+                    equivalence: "narrower",
+                  },
+                ],
+              }
+            : element
+        ),
+      })) ?? [],
+  };
+}
+
+async function processConceptMap(inStream: Readable): Promise<void> {
+  const rl = createInterface(inStream);
+  const client = new TerminologyClient();
+
+  // const counts = Object.create(null) as Record<string, number>;
+  // const codings = Object.create(null) as Record<string, Coding[]>;
+  const mappedConcepts: Record<string, ConceptMap> = Object.create(null);
+  for await (const line of rl) {
+    const concept = new UmlsConceptMap(line);
+    if (concept.MAPSETCUI != "C5885096") {
+      continue;
+    }
+    if (concept.MAPRULE && IFA_RULE_REGEX.test(concept.MAPRULE)) {
+      continue;
+    }
+    if (concept.REL != "RO") {
+      continue;
+    }
+
+    const key = concept.MAPSETCUI + "|" + concept.FROMID;
+
+    const sourceParameters = createLookupParameters(C5885096.system, concept.FROMID);
+    const sourceLookup = await client.lookupCode(sourceParameters);
+    const sourceDisplay = sourceLookup[0].display;
+
+    if (concept.TOID.endsWith("?")) {
```

Comment:
tldr here is that some icd10 codes end in a ? and those map to a range of different icd10 codes based on what the 7th character is. The 7th character is a kind of toggleable field in icd10 that indicates some attributes about the condition. So we need to lookup all the possible variation of the code and then map all of those